### ISSUE TITLE
Node._children doesn't sync to native in time on native.

### ISSUE THREAD

**Issue by minggo** [cocos/cocos-engine#15780]
In order to improve performance, some deserialization work is done in Typescript. Typescript will deserialize Node._children, then sync it c++. But c++ codes may access Node._children before Typescript synchronizing it to c++. Then it will cause crash.

https://github.com/cocos/cocos-engine/pull/15777 bypass the crash issue. Should revert it after this issue is fixed.

**Comment by minggo** [comment]
There is another issue caused by this issue. Currently, `Node.setSiblingIndex()` reuses `_siblingIndex` like this
```c++
void Node::setSiblingIndex(index_t index) {
    ...
    siblings.erase(siblings.begin() + _siblingIndex);
    ...
}
```
But `_siblingIndex` is always 0, the reason is mentioned above. So it will erase wrong child. If change to
```c++
void Node::setSiblingIndex(index_t index) {
    ...
    siblings.erase(siblings.begin() + getIdxOfChild(siblings, this));
    ...
}
```
Then it works. But it is not needed.

**Comment by dumganhar** [comment]
It's because setSiblingIndex is called too early and it's a bug in 3.8.0 and fixed in 3.8.1 here https://github.com/cocos/cocos-engine/pull/15776
